File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/context/roleplay/AtlasPointsInformations.scala





















// Generated on 05/03/2013 10:35:04
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait AtlasPointsInformations extends TypeDeserializer {
  val typeId = 175

  protected trait Type extends BaseType with Serializable {
    val typeId = AtlasPointsInformations.typeId

    val theType: Byte
    val coords: Seq[MapCoordinatesExtended.TypeImpl]

    def serialize(buf: Buffer) {
      buf.writeByte(theType)
      buf.writeUShort(coords.length)
      for (entry <- coords) {
        entry.serialize(buf)
      }
    }

  }

  protected def create(theType: Byte, coords: Seq[MapCoordinatesExtended.TypeImpl]): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val theType = buf.readByte()
    val coords = {
      val builder = collection.immutable.Seq.newBuilder[MapCoordinatesExtended.TypeImpl]
      for (i <- (0 to buf.readUShort())) {
        builder += MapCoordinatesExtended.deserialize(buf)
      }
      builder.result()
    }

    create(theType, coords)
  }

}

object AtlasPointsInformations extends AtlasPointsInformations {
  case class TypeImpl(theType: Byte, coords: Seq[MapCoordinatesExtended.TypeImpl]) extends Type
  override def create(theType: Byte, coords: Seq[MapCoordinatesExtended.TypeImpl]) = TypeImpl(theType, coords)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(theType: Byte, coords: Seq[MapCoordinatesExtended.TypeImpl]) = create(theType, coords)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/AtlasPointsInformations.java


// Generated on 05/08/2013 19:38:01
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class AtlasPointsInformations extends NetworkType {
    public static final short TYPE_ID = 175;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public byte type;
    public MapCoordinatesExtended[] coords;
    
    public AtlasPointsInformations() { }
    
    public AtlasPointsInformations(byte type, MapCoordinatesExtended[] coords) {
        this.type = type;
        this.coords = coords;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeByte(type);
        buf.writeUShort(coords.length);
        for (MapCoordinatesExtended entry : coords) {
            entry.serialize(buf);
        }
    }
    
    @Override
    public void deserialize(Buffer buf) {
        type = buf.readByte();
        if (type < 0)
            throw new RuntimeException("Forbidden value on type = " + type + ", it doesn't respect the following condition : type < 0");
        int limit = buf.readUShort();
        coords = new MapCoordinatesExtended[limit];
        for (int i = 0; i < limit; i++) {
            coords[i] = new MapCoordinatesExtended();
            coords[i].deserialize(buf);
        }
    }
    
}
